src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs

```diff
@@ -16,23 +18,30 @@ public class RedisCacheOptions : IOptions<RedisCacheOptions>
         /// <summary>
         /// The configuration used to connect to Redis.
         /// </summary>
-        public string Configuration { get; set; }
-        
+        public string? Configuration { get; set; }
+
         /// <summary>
         /// The configuration used to connect to Redis.
         /// This is preferred over Configuration.
         /// </summary>
-        public ConfigurationOptions ConfigurationOptions { get; set; }
+        public ConfigurationOptions? ConfigurationOptions { get; set; }
+
+        /// <summary>
+        /// Gets or sets a delegate to create the ConnectionMultiplexer instance.
+        /// </summary>
+        [DisallowNull]
+        public Func<Task<IConnectionMultiplexer>>? ConnectionMultiplexerFactory { get; set; }
 
         /// <summary>
         /// The Redis instance name.
         /// </summary>
-        public string InstanceName { get; set; }
+        public string? InstanceName { get; set; }
 
         /// <summary>
         /// The Redis profiling session
         /// </summary>
-        public Func<ProfilingSession> ProfilingSession { get; set; }
+        [DisallowNull]
```

Comment:
@pranavkm I added a [DisallowNull] here like ConnectionMultiplexerFactory. Thoughts? I originally merged the PR in https://github.com/dotnet/aspnetcore/pull/31018 and brought it to API review post-hoc. Turning on nullability now, it seems to make sense to add `DisallowNull` here too.

Reply:
Yup, makes sense.